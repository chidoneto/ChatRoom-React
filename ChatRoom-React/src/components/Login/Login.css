.Login {
    position: fixed;
    left: 0;
    bottom: 0;
    top: 0;
    padding: 0;
    margin: 0;
    width: 100%;
    height: 100%;
    background-color: white;
    z-index: 100;
    display: flex;
    flex-flow: column;
    justify-content: center;
    box-sizing: border-box;
}

.Login .input {
    width: 30%;
    margin: 15px auto;
    padding: 15px 30px;
    font-size: 100%;
    border: 1px solid #EFF1F2;
    border-radius: 5px;
    box-shadow: inset 0 0 3px rgb(80,80,80, 0.3);
}

.Login input:focus, textarea:focus, select:focus{
    outline: none;
}

.Login div {
    width: 30%;
    margin: 15px auto;
    padding: 15px 30px;
    font-size: 100%;
    font-weight: 500;
    border: 1px solid #FF1940;
    border-radius: 5px;
    color: white;
    background-color: #FF1940;
    -webkit-transition: background-color 0.3s, border 0.3s, color 0.3s;
    transition: background-color 0.3s, border 0.3s, color 0.3s;
}

.Login div:hover {
    border-radius: 5px;
    color: #FF1940;
    background-color: white;
}

@media (max-width: 480px) {
    .Login .input {
        width: 70%;
    }
    .Login div {
        width: 70%;
    }
}
