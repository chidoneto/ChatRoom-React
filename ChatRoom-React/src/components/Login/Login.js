import React, { useState, useRef, useEffect } from 'react';
import { PropTypes } from 'prop-types';

import './Login.css';


const Login = (props) => {
  const { onLogIn } = props;

  const [inputValue, setInputValue] = useState('');
  const inputRef = useRef();

  const submitHandler = (event) => {
    event.preventDefault();
    onLogIn(inputValue);
  };

  useEffect(() => {
    inputRef.current.focus();
  }, []);

  return (
    <form className="Login" onSubmit={submitHandler} >
        <input ref={inputRef}
          className="input"
          placeholder={"Type your username..."}
          value={inputValue}
          onChange={event => setInputValue(event.target.value)}
        >
        </input>
        <div className="button" onClick={submitHandler} >Join the DoorDash Chat!</div>
    </form>
  );
};

Login.propTypes = {
  onLogIn: PropTypes.func.isRequired
};

export default Login;