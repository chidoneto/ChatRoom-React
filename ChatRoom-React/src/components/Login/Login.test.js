import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './Login.js';
 
configure({adapter: new Adapter()});
 
describe('<Login /> form', () => {
  let userNameValue;
  const mockCallback = jest.fn((userName) => { userNameValue = userName; });
  let wrapper;
  
  beforeEach(() => { wrapper = shallow(<Login onLogIn={mockCallback}/>); });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an input and a button', () => {
    expect(wrapper.find('.Login')).toHaveLength(1);
    expect(wrapper.find('.input')).toHaveLength(1);
    expect(wrapper.find('.button')).toHaveLength(1);
  });

  it('should trigger callback on button click', () => {
    wrapper.find('.button').simulate('click');
    expect(mockCallback).toHaveBeenCalledTimes(1);
  });

  it('should set user name, and receive such value back on click', () => {
    const USER_NAME = 'Joe Doe';

    wrapper.find('.input').simulate('change', { target: { value: USER_NAME } });
    expect(wrapper.find('.input').prop('value')).toEqual(USER_NAME);
    wrapper.find('.button').simulate('click');
    expect(userNameValue).toEqual(USER_NAME);
  });
});
