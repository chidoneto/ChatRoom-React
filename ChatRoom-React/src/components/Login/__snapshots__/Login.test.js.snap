// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> form should match the snapshot 1`] = `ShallowWrapper {}`;
